src/common/cockpitwebcertificate.c

```diff
@@ -111,6 +111,41 @@ cockpit_certificate_locate (char **error)
   return NULL;
 }
 
+/**
+ * cockpit_certificate_key_path:
+ *
+ * Check if there is a matching *.key file for the given *.cert file.
+ *
+ * Return key file path on success, or %NULL if there is no matching *.key file.
+ */
+char *
+cockpit_certificate_key_path (const char *certfile)
+{
+  size_t len = strlen (certfile);
+  char *keypath = NULL;
+
+  /* invalid if it does not end with .cert */
+  if (len <= 5 || strcmp (certfile + len - 5, ".cert") != 0)
+    {
+      warnx ("invalid certificate file name: %s", certfile);
+      return NULL;
+    }
+
+  keypath = strdupx (certfile);
```

Comment:
Better use `asprintf` with `%.*s` here.

Reply:
Done, thanks for the hint! This is so much nicer.